src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,90 @@
+using Flux
+
+basicblock(inplanes::Int, outplanes::Int, downsample::Bool = false) = downsample ? 
+  Chain(Conv((3, 3), inplanes => outplanes, stride = 1, pad = 1),
+        BatchNorm(outplanes, λ = relu),
+        Conv((3, 3), outplanes => outplanes, stride = 2, pad = 1,
+        BatchNorm(outplanes, λ=relu)) : 
+  Chain(Conv((3, 3), inplanes => outplanes, stride = 1, pad = 1),
+        BatchNorm(outplanes, λ = relu),
+        Conv((3, 3), outplanes => outplanes, stride = 1, pad = 1,
+        BatchNorm(outplanes, λ=relu))
+
+bottleneck(inplanes::Int, outplanes::Int, downsample::Bool = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outplanes, stride = 2),
+        BatchNorm(outplanes, λ = relu)) :
+  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
+        BatchNorm(inplanes, λ = relu),
+        Conv((1, 1), inplanes => outplanes, stride = 1),
+        BatchNorm(outplanes, λ = relu))
+
+projection(inplanes::Int, planes::Int, stride::Int) = Chain(Conv((1, 1), inplanes => outplanes, stride=stride),
+                                                            BatchNorm((outplanes), λ=relu))
+
+
+function resnet(channel_config::AbstractArray{Int,1}, block_config::AbstractArray{Int,1})
+   # begin
+   # planes = 64
+   # inplanes = 64
+   # expansion = 1
+   # stride = 1
+   # end
+    layers = []
+    push!(layers, Conv((7, 7), 3=>inplanes, stride=(2, 2), pad=(3, 3)))
+    push!(layers, BatchNorm(inplanes, λ=relu))
+    push!(layers, MaxPool((3, 3), stride=(2, 2), pad=(1, 1)))
+    inplanes = 64
+    for nrepeats in block_config
+      if length(channel_config) == 2
```

Comment:
Instead of checking the `length(channel_config)`, we should allow for more flexible construction. I would remove the two `if`-blocks. They both do the same thing, so just keep the body of one of them. Then add arguments `block` and `shortcut` to the `resnet` function. Then, our loop body should look like:
```julia
push!(layers, SkipConnection(block(inplanes, outplanes, i == 1), shortcut(inplanes, outplanes, stride)))
```
This will allow users to pass in either `basicblock` or `bottleneck` as necessary. Same for `shortcut` which can be `projection` or `identity`. You will need to define the following `identity` function:
```julia
identity(inplanes, outplanes, stride) = +
```